splicemachine/spliceengine
# DB-11603 Properly separate privileges drop when dropping roles

## Short Description
DataDictionary.dropAllPermsByGrantee should filter grantees per databaseId

## Long Description
In a multidatabase environment, only checking for grantees is not enough anymore when attempting to drop privileges. We should filter privileges by databaseId before dropping

## How to test
*splicedb2*:
```
CREATE SCHEMA S; CREATE ROLE FOO; GRANT ALL PRIVILEGES ON SCHEMA S TO FOO;
select * from sysvw.sysschemapermsview;
SCHEMAPER&|GRANTEE   |GRANTOR   |SCHEMAID  |&|&|&|&|&|&|MOD&|ACC&|SCHEMANAME
60430428-&|FOO       |SPLICE    |a23943da-&|y|y|y|y|y|y|y   |y   |S
```

*splicedb*:
```
CREATE SCHEMA S; CREATE ROLE FOO; GRANT ALL PRIVILEGES ON SCHEMA S TO FOO;
select * from sysvw.sysschemapermsview;
SCHEMAPER&|GRANTEE   |GRANTOR   |SCHEMAID  |&|&|&|&|&|&|MOD&|ACC&|SCHEMANAME
97fc8455-&|FOO       |SPLICE    |bbe583e4-&|y|y|y|y|y|y|y   |y   |S
```

Now drop one role FOO:

*splicedb*:
```
DROP ROLE FOO;
select * from sysvw.sysschemapermsview;
SCHEMAPER&|GRANTEE   |GRANTOR   |SCHEMAID  |&|&|&|&|&|&|MOD&|ACC&|SCHEMANAME
```

The privileges on *splicedb2* should not be affected:

*splicedb2*:
```
select * from sysvw.sysschemapermsview;
SCHEMAPER&|GRANTEE   |GRANTOR   |SCHEMAID  |&|&|&|&|&|&|MOD&|ACC&|SCHEMANAME
60430428-&|FOO       |SPLICE    |a23943da-&|y|y|y|y|y|y|y   |y   |S
```
